pkg/kn/commands/revision/delete.go

```diff
@@ -47,6 +62,21 @@ func NewRevisionDeleteCommand(p *commands.KnParams) *cobra.Command {
 			if err != nil {
 				return err
 			}
+			// Create list filters
+			var params []v1.ListConfig
+			if prune {
+				params = append(params, v1.WithService(pruneFilter))
+			}
+			if prune || pruneAll {
+				args, err = getUnreferencedRevisionNames(params, client)
+				if err != nil {
+					return err
+				}
+				if len(args) == 0 {
+					fmt.Fprintf(cmd.OutOrStdout(), "No unreferenced revisions found.\n")
+					return nil
+				}
+			}
```

Comment:
we can create the client first and then replace the multiple if/else with `switch`